openedx/core/djangoapps/oauth_dispatch/api.py

```diff
@@ -33,7 +33,9 @@ def create_dot_access_token(request, user, client, expires_in=None, scopes=None)
         request_validator=dot_settings.OAUTH2_VALIDATOR_CLASS(),
     )
     _populate_create_access_token_request(request, user, client, scopes)
-    return token_generator.create_token(request, refresh_token=True)
+    token = token_generator.create_token(request, refresh_token=True)
+    token_generator.request_validator.save_token(token, request)
```

Comment:
The `create_token` method was also saving token in the database in the previous OAuth lib (oauthlib==3.0.1) version which we were using.
https://github.com/oauthlib/oauthlib/blob/v3.0.1/oauthlib/oauth2/rfc6749/tokens.py#L341.

But in the latest version `save_token` has been deprecated.
https://github.com/oauthlib/oauthlib/blob/v3.2.2/oauthlib/oauth2/rfc6749/tokens.py#L303

Now we need to save tokens explicitly.

**PR in which save_token functionaly was removed from oauthlib** [PR-653](https://github.com/oauthlib/oauthlib/pull/653/files#diff-0929b230a59e3bb98cdf0a5433fdc4b5c5baf783284ed34a7ad896ffa6478d3dL340)

Reply:
1. I am guessing that the refresh token must be saved either way. Is that correct?
2. I am guessing that we don't need to save JWT tokens, but I'm not certain. I don't think you should spend time working on this, but does the following comment make sense to add?:
```
# We may not need to save our JWT tokens, but we do need to save the opaque Bearer tokens.
# However, since BearerAuthentication is deprecated, we may want to explore removing this save
# call in the future once the DEPR has been completed.
# See https://github.com/openedx/edx-drf-extensions/issues/284
```

Reply:
1. I guess `refresh_token` is independent of the change that we're talking about here.
As you can see refresh token implementation in both versions is similar.

|[v3.2.2 implementaiton](https://github.com/oauthlib/oauthlib/blob/v3.2.2/oauthlib/oauth2/rfc6749/tokens.py#L327-L332)|[v3.0.1 implementation](https://github.com/oauthlib/oauthlib/blob/v3.0.1/oauthlib/oauth2/rfc6749/tokens.py#L331-L336)|
|---|---|

Reply:
(2) 
Yeah, it makes sense that if we're moving towards JWT we can skip the save call.

But there are other things to consider:
- Although this OEP [(oep-42)](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0042-bp-authentication.html#oauth2-and-bearer-tokens:~:text=BearerAuthentication%20(edx%2Ddrf%2Dextensions)) states that ` BearerAuthentication (edx-drf-extension)` is deprecated, but it's still there [(link to BearerAuthentication 
 class)](https://github.com/openedx/edx-drf-extensions/blob/e8c7e0c743cc1bca6c89953f1cd94badfaaad99c/edx_rest_framework_extensions/auth/bearer/authentication.py#L18)

- Also this PR is in `edx-platform`  where we have another quite similar`BearerAuthentication` class [(link to class)](https://github.com/openedx/edx-platform/blob/f20b7ec985e9a5450b48b9725c4f76821041e844/openedx/core/lib/api/authentication.py#L23). This class and its subclasses are being widely used in the code base [link to usage](https://github.com/search?q=repo%3Aopenedx%2Fedx-platform%20BearerAuthentication&type=code) 

So final thoughts are, as long as the `BearerAuthentication` is there, we need the `save call`.

Reply:
1. Thank you for the note on the edx-platform BearerAuthentication classes. That was a surprise to me. I have updated the DEPR with the additional information. The point of that DEPR is to remove all of BearerAuthentication, so I simply missed these implementation details.
2. What do you think of the following updated comment proposal:
```
# This save_token call is required with BearerAuthentication. Once the DEPR for
# BearerAuthentication is complete and it has been fully removed, we may no
# longer need to save the token, since JWT tokens don't rely on the database.
# See DEPR https://github.com/openedx/edx-drf-extensions/issues/284
```

Additionally, or alternatively, once this lands we can add a comment in the DEPR ticket pointing to the save_token call, and note the same thing.